core/lib/Thelia/Action/Product.php

```diff
@@ -100,6 +121,451 @@ public function create(ProductCreateEvent $event)
         $event->setProduct($product);
     }
 
+    /*******************
+     * CLONING PROCESS *
+     *******************
+     *
+     * Vars prefixes :
+     * op = Original Product
+     * cp = Clone Product
+     */
+
+    /**
+     * @param ProductCloneEvent $event
+     */
+    public function cloneProduct(ProductCloneEvent $event)
+    {
+        // Get important datas
+        $lang = $event->getLang();
+        $opId = $event->getProductId();
+
+        $op = ProductQuery::create()
+            ->findOneById($opId);
+
+        $opDefaultI18n = ProductI18nQuery::create()
+            ->filterByLocale($lang, Criteria::EQUAL)
+            ->findOneById($opId);
+
+        $opDefaultPrice = ProductPriceQuery::create()
+            ->findOneByProductSaleElementsId($op->getDefaultSaleElements()->getId());
+
+        $opPSEs = ProductSaleElementsQuery::create()
+            ->orderByIsDefault(Criteria::DESC)
+            ->findByProductId($opId);
+
+        // Cloning process
+
+        $cp = $this->createClone($op, $opDefaultI18n, $opDefaultPrice, $event);
```

Comment:
Once the new product created, remove the PSE and manage all PSE in the foreach line 169